src/Microsoft.Sbom.Api/Workflows/SBOMRedactionWorkflow.cs

```diff
@@ -17,17 +24,133 @@ public class SbomRedactionWorkflow : IWorkflow<SbomRedactionWorkflow>
+                {
+                    await JsonSerializer.SerializeAsync(outStream, redactedSpdx);
+                }
+
+                log.Information($"Redacted SBOM {sbomPath} saved to {outputPath}");
+            }
+            catch (Exception ex)
+            {
+                throw new Exception($"Failed to redact {sbomPath}: {ex.Message}", ex);
+            }
+        }
+
+        return true;
+    }
+
+    private async Task<IDictionary<string, IValidatedSBOM>> GetValidSbomsAsync()
+    {
+        var sbomPaths = GetInputSbomPaths();
+        var validatedSboms = new Dictionary<string, IValidatedSBOM>();
+        foreach (var sbom in sbomPaths)
+        {
+            log.Information($"Validating SBOM {sbom}");
+            var validatedSbom = validatedSBOMFactory.CreateValidatedSBOM(sbom);
+            var validationDetails = await validatedSbom.GetValidationResults();
+            if (validationDetails.Status != FormatValidationStatus.Valid)
+            {
+                throw new InvalidDataException($"Failed to validate {sbom}:\n{string.Join('\n', validationDetails.Errors)}");
+            }
+            else
+            {
+                validatedSboms.Add(sbom, validatedSbom);
+            }
+        }
+
+        return validatedSboms;
+    }
+
+    private string GetOutputPath(string sbomPath)
+    {
+        return fileSystemUtils.JoinPaths(configuration.OutputPath.Value, fileSystemUtils.GetFileName(sbomPath));
+    }
+
+    private IEnumerable<string> GetInputSbomPaths()
+    {
+        if (configuration.SbomPath?.Value != null)
+        {
+            return new List<string>() { configuration.SbomPath.Value };
+        }
+        else if (configuration.SbomDir?.Value != null)
+        {
+            return fileSystemUtils.GetFilesInDirectory(configuration.SbomDir.Value);
+        }
+        else
+        {
+            throw new Exception("No valid input SBOMs to redact provided.");
+        }
+    }
+
+    private string ValidateDirStrucutre()
+    {
+        string inputDir;
+        if (configuration.SbomDir?.Value != null && fileSystemUtils.DirectoryExists(configuration.SbomDir.Value))
+        {
+            inputDir = configuration.SbomDir.Value;
+        }
+        else if (configuration.SbomPath?.Value != null && fileSystemUtils.FileExists(configuration.SbomPath.Value))
+        {
+            inputDir = fileSystemUtils.GetDirectoryName(configuration.SbomPath.Value);
+        }
+        else
+        {
+            throw new ArgumentException("No valid input SBOMs to redact provided.");
+        }
+
+        var outputDir = configuration.OutputPath.Value;
+        if (fileSystemUtils.GetFullPath(outputDir).Equals(fileSystemUtils.GetFullPath(inputDir)))
+        {
+            throw new ArgumentException("Output path cannot be the same as input SBOM directory.");
+        }
+
+        if (!fileSystemUtils.DirectoryExists(outputDir))
+        {
+            fileSystemUtils.CreateDirectory(outputDir);
```

Comment:
Assuming `CreateDirectory` fails (bad drive, no permissions, no disk space, etc.), is the default Exception handling enough or should we provide a nicer error message?

Reply:
IMO default error handling is enough- any exceptions thrown here are caught higher up in the code and wrapped in a cleaner error message